Odata-docs/odatalib/v7/edm/iedmtargetpath-for-annotations-target.md

```diff
@@ -0,0 +1,463 @@
+            this.Customer.AddKeys(this.Customer.AddStructuralProperty("Id", EdmCoreModel.Instance.GetInt32(false)));
+            this.Customer.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            this.Customer.AddStructuralProperty("Address", new EdmComplexTypeReference(this.Address, false));
+            this.Model.AddElement(this.Customer);
+
+            this.VipCustomer = new EdmEntityType("NS", "VipCustomer", this.Customer);
+            this.Model.AddElement(this.VipCustomer);
+            this.City = new EdmEntityType("NS", "City");
+            this.City.AddStructuralProperty("Name", EdmPrimitiveTypeKind.String, isNullable: false);
+            this.Model.AddElement(this.City);
+
+            this.NavUnderComplex = this.Address.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "City",
+                    Target = this.City,
+                    TargetMultiplicity = EdmMultiplicity.One,
+                });
+
+            this.NavUnderCustomer = this.Customer.AddUnidirectionalNavigation(
+                new EdmNavigationPropertyInfo()
+                {
+                    Name = "Cities",
+                    Target = this.City,
+                    TargetMultiplicity = EdmMultiplicity.Many,
+                });
+
+            this.NameProperty = this.Customer.DeclaredProperties.Where(x => x.Name == "Name").FirstOrDefault();
+            this.AddressProperty = this.Customer.DeclaredProperties.Where(x => x.Name == "Address").FirstOrDefault();
+
+            this.Container = new EdmEntityContainer("NS", "Default");
+            this.EntitySet = new EdmEntitySet(this.Container, "Customers", this.Customer);
+            this.Singleton = new EdmSingleton(this.Container, "Me", this.Customer);
+            this.Container.AddElement(this.EntitySet);
+            this.Container.AddElement(this.Singleton);
+
+            this.Model.AddElement(this.Container);
+        }
+
+        public EdmModel Model { get; private set; }
+        public EdmEntityContainer Container { get; private set; }
+        public EdmEntitySet EntitySet { get; private set; }
+        public EdmSingleton Singleton { get; private set; }
+        public EdmEntityType Customer { get; private set; }
+        public EdmEntityType VipCustomer { get; private set; }
+        public EdmEntityType City { get; private set; }
+        public EdmComplexType Address { get; private set; }
+        public EdmComplexType DerivedAddress { get; private set; }
+        public IEdmProperty NameProperty { get; private set; }
+        public IEdmProperty AddressProperty { get; private set; }
+        public EdmNavigationProperty NavUnderComplex { get; private set; }
+        public EdmNavigationProperty NavUnderCustomer { get; private set; }
+    }
+}
+
+```
+
+## Write out-of-line annotations
+
+Replace the contents of _Program.cs_ file with the following code:
```

Comment:
At least sentences to illustrate what will do in this session? 
Or what you want to achieve? Use a 'concrete' scenario could be better.

for example:  I want to limit the 'user' access for 'customers/{key}/detailInfo'? 
How can we use the annotations on this path to specify the 'allowed' user accesses?